### PR TITLE
[system test] [doc] connect package

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Enhancement / new feature
- Refactoring
- Documentation

### Description

Similarly as in [1] PR. In this one, I have added a system test doc to our connect tests.

[1] - https://github.com/strimzi/strimzi-kafka-operator/pull/10225

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

File: development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+| Step | Action | Result |
+| - | - | - |
+| 1. | Create TestStorage object | TestStorage instance is created with context |
+| 2. | Get image name for test case | Image name is successfully retrieved |
+| 3. | Create Kafka Topic resources | Kafka Topic resources are created with wait |
+| 4. | Create Kafka Connect resources | Kafka Connect resources are created with wait |
+| 5. | Configure Kafka Connector | Kafka Connector is configured and created with wait |
+| 6. | Verify Kafka Connector class name | Connector class name matches expected ECHO_SINK_CLASS_NAME |
+| 7. | Create Kafka Clients and send messages | Kafka Clients created and messages sent and verified |
+| 8. | Check logs for received message | Logs contain the expected received message |
```
Comment (henryZrncik) [inline development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md:93]:
```suggestion
| 8. | Check logs for received message in connector pod | Logs contain the expected received message |
```

**Review by henryZrncik — APPROVED** [review]
just nits thanks!

**Review by PaulRMellor — APPROVED** [review]
Test descriptions are clear :+1:

File: development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+## testBuildFailsWithWrongChecksumOfArtifact
+
+**Description:** Test that ensures Kafka Connect build fails with wrong artifact checksum and recovers with correct checksum.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Initialize TestStorage and get test image name | TestStorage instance is created and the image name for the test case is retrieved |
```
[the remark this answers is not in the document]
Comment (see-quick) [reply development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md:25]: I don't have any strong opinion on this.  Up to us @strimzi/system-test-contributors .

[the remark this answers is not in the document]
Comment (im-konge) [reply development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md:25]: I would maybe remove the part about creating TestStorage -> as it is done in every testcase.

File: development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+## testBuildOtherPluginTypeWithAndWithoutFileName
+
+**Description:** Test verifying Kafka Connect plugin behavior with and without file names for different plugin types.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Initialize test storage and create namespace and topic | Namespace and topic are created successfully |
```
[the remark this answers is not in the document]
Comment (im-konge) [reply development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md:47]: Yeah I agree here as well. Let's stick just with the steps that are done in the testcase itself

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java [hunk 4]
```diff
@@ -304,6 +362,22 @@ void testPushIntoImageStream() {
...
 
     @ParallelTest
+    @TestDoc(
+        description = @Desc("Test updating and validating Kafka Connect with another plugin."),
+        steps = {
+            @Step(value = "Create TestStorage instance", expected = "Instance of TestStorage is created"),
+            @Step(value = "Generate random topic name and create Kafka topic", expected = "Kafka topic is successfully created"),
+            @Step(value = "Deploy Kafka Connect and Scraper pod with specific configurations", expected = "Kafka Connect and Scraper pod are successfully deployed"),
+            @Step(value = "Deploy network policies for KafkaConnect", expected = "Network policies are successfully deployed"),
+            @Step(value = "Create and validate EchoSink KafkaConnector", expected = "EchoSink KafkaConnector is successfully created and validated"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java:372]: What does "validate" here means? How is it validated?

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java [hunk 5]
```diff
@@ -304,6 +362,22 @@ void testPushIntoImageStream() {
...
+            @Step(value = "Add a second plugin to Kafka Connect and perform rolling update", expected = "Second plugin is added and rolling update is performed"),
+            @Step(value = "Create and validate Camel-HTTP-Sink KafkaConnector", expected = "Camel-HTTP-Sink KafkaConnector is successfully created and validated"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java:374]: Same here

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java [hunk 6]
```diff
@@ -398,6 +472,19 @@ void testUpdateConnectWithAnotherPlugin() {
     }
 
     @ParallelTest
+    @TestDoc(
+        description = @Desc("Test verifying Kafka Connect plugin behavior with and without file names for different plugin types."),
+        steps = {
+            @Step(value = "Initialize test storage and create namespace and topic", expected = "Namespace and topic are created successfully"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java:478]: Isn't the Namespace already created before?

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java [hunk 7]
```diff
@@ -457,6 +544,20 @@ void testBuildOtherPluginTypeWithAndWithoutFileName() {
     @Tag(ACCEPTANCE)
     @KindNotSupported("using kind we encounter (error building image: deleting file system after stage 0: unlinkat //product_uuid: device or resource busy)")
     @ParallelTest
+    @TestDoc(
+        description = @Desc("Test building a plugin using Maven coordinates artifacts."),
+        steps = {
+            @Step(value = "Create a test storage object", expected = "Test storage object is created"),
+            @Step(value = "Generate image name for the test case", expected = "Image name is generated successfully"),
+            @Step(value = "Create Kafka topic and Kafka Connect resources", expected = "Resources are created and available"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java:552]: I would write here what configuration of KafkaConnect you are providing here (the configuration for plugin using mvn coordinates)

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 8]
```diff
@@ -106,11 +111,36 @@
 @Tag(CONNECT)
 @Tag(CONNECT_COMPONENTS)
 @SuppressWarnings({"checkstyle:ClassDataAbstractionCoupling"})
+@SuiteDoc(
+    description = @Desc("Verifies the deployment, manual rolling update, and undeployment of Kafka Connect components."),
+    beforeTestSteps = {
+        @Step(value = "Deploy uber operator across all namespaces, with custom configuration", expected = "Uber operator is deployed"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:117]: uber operator is a bit misleading in this case IMHO
I would just use Cluster Operator or something

uber operator, for me, is operator deploying other operators, which deploy another components.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 9]
```diff
@@ -106,11 +111,36 @@
...
+        @Step(value = "Deploy management Pod for accessing all other Pods", expected = "Management Pod is deployed")
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:118]: Is is scraper Pod? So maybe something like
```suggestion
        @Step(value = "Deploy scraper Pod for accessing all other Pods", expected = "Scraper Pod is deployed")
```

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 10]
```diff
@@ -106,11 +111,36 @@
...
     @ParallelNamespaceTest
+    @TestDoc(
+        description = @Desc("Verifies the deployment, manual rolling update, and undeployment of Kafka Connect components."),
+        steps = {
+            @Step(value = "Initialize Test Storage", expected = "Test storage instance is created with required context"),
+            @Step(value = "Define expected configurations", expected = "Configurations are loaded from properties file"),
+            @Step(value = "Create and wait for resources", expected = "Kafka resources, including NodePools and KafkaConnect instances, are created and become ready"),
+            @Step(value = "Annotate for manual rolling update", expected = "KafkaConnect components are annotated for a manual rolling update"),
+            @Step(value = "Perform and wait for rolling update", expected = "KafkaConnect components roll and new pods are deployed"),
+            @Step(value = "Verify Kafka Connect pod", expected = "Pod configurations and annotations are verified"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:137]: Here as well - what does verify mean? I see it in the expected part, but I would mention it in the `value`

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 11]
```diff
@@ -106,11 +111,36 @@
...
+            @Step(value = "Verify connectors", expected = "Various Kafka Connect resource labels and configurations are verified to ensure correct deployment")
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:138]: Same here

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 12]
```diff
@@ -232,6 +256,24 @@ void testKafkaConnectAndConnectorStateWithFileSinkPlugin() {
     }
 
     @ParallelNamespaceTest
+    @TestDoc(
+        description = @Desc("Test verifying Kafka Connect functionalities with Plain and SCRAM-SHA authentication."),
+        steps = {
+            @Step(value = "Create object instance of TestStorage", expected = "Instance of TestStorage is created"),
+            @Step(value = "Create NodePools using resourceManager", expected = "NodePools for broker and controller are created"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:263]: The NodePools are created in each of the testcase (or not, based on configuration). Should this step be anywhere else?

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 13]
```diff
@@ -306,6 +348,22 @@ void testKafkaConnectWithPlainAndScramShaAuthentication() {
...
+        steps = {
+            @Step(value = "Create and configure test storage", expected = "Test storage is set up with necessary configurations."),
+            @Step(value = "Create and wait for the broker and controller pools", expected = "Broker and controller pools are created and running."),
+            @Step(value = "Deploy and configure Kafka Connect with File Sink Plugin", expected = "Kafka Connect with File Sink Plugin is deployed and configured."),
+            @Step(value = "Deploy Network Policies for Kafka Connect", expected = "Network Policies are successfully deployed for Kafka Connect."),
+            @Step(value = "Create and wait for Kafka Connector", expected = "Kafka Connector is created and running."),
+            @Step(value = "Deploy and configure scraper pod", expected = "Scraper pod is deployed and configured."),
+            @Step(value = "Deploy and configure Kafka clients", expected = "Kafka clients are deployed and configured."),
+            @Step(value = "Execute assertions to verify the Kafka Connector", expected = "Assertions pass, confirming that the Kafka Connector is working as expected.")
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:361]: The other tests have also assertions, I think that you can just have there "verify that Kafka Connector has ..." or something like that. As I mentioned in other comments, having just "verify Kafka Connector" is not saying much. What is verified? This applies to other testcases as well

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 14]
```diff
@@ -503,6 +607,26 @@ void testSecretsWithKafkaConnectWithTlsAndTlsClientAuthentication() {
     }
 
     @ParallelNamespaceTest
+    @TestDoc(
+        description = @Desc("Test validating Kafka Connect with TLS and SCRAM-SHA authentication along with associated resources setup and verification."),
+        steps = {
+            @Step(value = "Initialize test storage and resource manager", expected = "Instances created successfully"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:613]: Isn't resource manager initialized before? Also, I guess that it's not needed to mention it there (as it doesn't have much value in the whole test steps IMHO).

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 15]
```diff
@@ -822,6 +993,19 @@ void testMultiNodeKafkaConnectWithConnectorCreation() {
 
     @Tag(CONNECTOR_OPERATOR)
     @ParallelNamespaceTest
+    @TestDoc(
+        description = @Desc("Test verifying Kafka connect TLS authentication with a username containing unusual characters."),
+        steps = {
+            @Step(value = "Create a user with a username containing dots and 64 characters", expected = "User is created successfully"),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:999]: There is no user created, you just create a variable with username

**Review by im-konge — APPROVED** [review]
Thanks for this, I left some comments.

File: development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (see-quick) [reply development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md:47]: Updated

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java [hunk 16]
```diff
@@ -304,6 +362,22 @@ void testPushIntoImageStream() {
     }
 
     @ParallelTest
+    @TestDoc(
+        description = @Desc("Test updating and validating Kafka Connect with another plugin."),
+        steps = {
+            @Step(value = "Create TestStorage instance", expected = "Instance of TestStorage is created"),
+            @Step(value = "Generate random topic name and create Kafka topic", expected = "Kafka topic is successfully created"),
+            @Step(value = "Deploy Kafka Connect and Scraper pod with specific configurations", expected = "Kafka Connect and Scraper pod are successfully deployed"),
```
[the remark this answers is not in the document]
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java:370]: I have metioned it also in `testBuildFailsWithWrongChecksumOfArtifact`

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java (hunk 4 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java:372]: That is already written in the `expected` scope so I will remove this.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java (hunk 6 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java:478]: true

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java (hunk 8 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:117]: Removed all before test steps here.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java (hunk 12 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:263]: True will update this.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java (hunk 14 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:613]: True updated

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java (hunk 7 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectBuilderST.java:549]: Thanks 👍

File: development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (see-quick) [reply development-docs/systemtests/io.strimzi.systemtest.connect.ConnectBuilderST.md:19]: Removed

**Pull request merged**